UNITTESTS/stubs/CMakeLists.txt

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) 2021 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+add_library(mbed-stubs-headers INTERFACE)
+add_library(mbed-headers INTERFACE)
+add_library(mbed-headers-base INTERFACE)
+add_library(mbed-headers-platform INTERFACE)
+add_library(mbed-headers-connectivity INTERFACE)
+add_library(mbed-headers-storage INTERFACE)
+add_library(mbed-headers-drivers INTERFACE)
+add_library(mbed-headers-hal INTERFACE)
+add_library(mbed-headers-events INTERFACE)
+add_library(mbed-headers-rtos INTERFACE)
```

Comment:
I think the names are a bit confusing - they are for unit tests only.
What about just a single `mbed-stubs-headers`:
```cmake
add_library(mbed-stubs-headers INTERFACE)

# platform
target_include_directories(mbed-stubs-headers
    INTERFACE
        ...
)


# storage
target_include_directories(mbed-stubs-headers
    INTERFACE
        ...
)

...
```
as CMake allows us to call `target_include_directories()` as many times as we need on one library.

Reply:
Or, a more major question: Is it possible for each test suite to directly link the original library (e.g. `mbed-storage`, `mbed-events, etc.)? Any source files that require real Mbed targets excluded in a similar way to aad3974ad5ab5472f34643915e210d280715785e.

(Personally I'm open for discussion, as we also don't want to overcomplicate stuff and this PR is already in a working state.)

Reply:
Unit tests need to be able to include headers selectively, for example only storage headers - not as a single library that contains all of them. You can of course remove this from mbed-os. I will then have to include my headers by hand in my tests or recreate these libs outside of mbed-os.

Reply:
> Unit tests need to be able to include headers selectively, for example only storage headers

It's not currently the case in this PR - it consolidates all `mbed-header-<component>` into `mbed-headers`, and links the latter to all tests.

Reply:
Nothing gets linked here, they are all interface libraries. The mbed-headers is for convenience: if you need all headers you just require it instead of listing all the individual ones. If you only want a subset you can require the individual components. Best of both worlds. My unit tests use that.

Reply:
> The mbed-headers is for convenience

This is what I meant by "link" (in CMake sense `target_link_libraries`).

Reply:
Including the original library will bring additional source c/cpp files into the build, ideally, unittests should list the required sources (unit under test and test code ) but I haven't tried to use direct original libraries.

Reply:
Okay the way it is, I think